src/test/java/org/minions/devfund/danielmontecinos/SquarelotronTest.java

```diff
@@ -0,0 +1,222 @@
+        };
+        Squarelotron actualResult = squarelotron.mainDiagonalFlip(ring);
+        assertArrayEquals(expectedResult, actualResult.getSquarelotronMatrix());
+    }
+
+    /**
+     * Tests mainDiagonalFlip method using six dimension array and ring equal to one.
+     */
+    @Test
+    public void testMainDiagonalFlipWithSixMatrixWithRingTwo() {
+        final int ring = 2;
+        final int dimension = 6;
+        final Squarelotron squarelotron = new Squarelotron(dimension);
+        final int[][] expectedResult = new int[][]{
+                {1, 2, 3, 4, 5, 6},
+                {7, 8, 14, 20, 26, 12},
+                {13, 9, 15, 16, 27, 18},
+                {19, 10, 21, 22, 28, 24},
+                {25, 11, 17, 23, 29, 30},
+                {31, 32, 33, 34, 35, 36}
+        };
+        Squarelotron actualResult = squarelotron.mainDiagonalFlip(ring);
+        assertArrayEquals(expectedResult, actualResult.getSquarelotronMatrix());
+    }
+
+    /**
+     * Tests rotateClockwise method using four dimension array and turns equal to one.
+     */
+    @Test
+    public void testRotateClockwiseFourDimensionMatrixOnce() {
+        final int dimension = 4;
+        final int turns = 1;
+        final Squarelotron squarelotron = new Squarelotron(dimension);
+        final int[][] expectedResult = new int[][]{
+                {13, 9, 5, 1},
+                {14, 10, 6, 2},
+                {15, 11, 7, 3},
+                {16, 12, 8, 4}};
+        squarelotron.rotateRight(turns);
+        assertArrayEquals(expectedResult, squarelotron.getSquarelotronMatrix());
+    }
+
+    /**
+     * Tests rotateClockwise method using two dimension array and turns equal to two.
+     */
+    @Test
+    public void testRotateAntiClockwiseTwoDimensionMatrixTwice() {
+        final int dimension = 2;
+        final int turns = -2;
+        final Squarelotron squarelotron = new Squarelotron(dimension);
+        final int[][] expectedResult = new int[][]{
+                {4, 3},
+                {2, 1}
+        };
+        squarelotron.rotateRight(turns);
+        assertArrayEquals(expectedResult, squarelotron.getSquarelotronMatrix());
+    }
+
+    /**
+     * Tests rotateClockwise method using three dimension array and turns equal to zero.
+     */
+    @Test
+    public void testRotateClockwiseThreeDimensionMatrixWithNoTurns() {
+        final int dimension = 3;
+        final int turns = 0;
+        final Squarelotron squarelotron = new Squarelotron(dimension);
+        final int[][] expectedResult = new int[][]{
+                {1, 2, 3},
+                {4, 5, 6},
+                {7, 8, 9}
+        };
+        squarelotron.rotateRight(turns);
+        assertArrayEquals(expectedResult, squarelotron.getSquarelotronMatrix());
+    }
+
+    /**
+     * Tests rotateClockwise method using three dimension array and turns equal to four.
+     */
+    @Test
+    public void testRotateClockwiseWithThreeDimensionMatrixWithAFourTurns() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testRotateClockwiseThreeDimensionMatrixWithNoTurns" on line 159. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)